File: src/main/java/dev/slne/surf/essentials/commands/cheat/gui/GrindstoneCommand.java
package dev.slne.surf.essentials.commands.cheat.gui;

import dev.slne.surf.essentials.utils.permission.Permissions;

public class GrindstoneCommand extends GuiCommand {
    GrindstoneCommand() {
        super("grindstone", "Der Schleifstein", Permissions.GRINDSTONE_SELF_PERMISSION, Permissions.GRINDSTONE_OTHER_PERMISSION);
    }

    @Override
    public void open(org.bukkit.entity.Player player) {
        player.openGrindstone(player.getLocation(), true);
    }
}

File: src/main/kotlin/dev/slne/surf/essentials/command/table/GrindstoneCommand.kt
@file:Suppress("DEPRECATION")

package dev.slne.surf.essentials.command.table

import dev.jorel.commandapi.kotlindsl.commandTree
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.slne.surf.api.core.messages.adventure.sendText
import dev.slne.surf.essentials.util.permission.EssentialsPermissionRegistry

fun grindstoneCommand() = commandTree("grindstone") {
    withPermission(EssentialsPermissionRegistry.GRINDSTONE_COMMAND)
    playerExecutor { player, _ ->
        player.openGrindstone(null, true)
        player.sendText {
            appendSuccessPrefix()
            success("Du hast einen Schleifstein geöffnet.")
        }
    }
}